sw/device/lib/base/log.c

```diff
@@ -64,11 +50,20 @@ void base_log_internal_core(log_severity_t severity, const char *file_name,
  * Logs |format| and the values that following in an efficient, DV-testbench
  * specific way.
  *
- * @param severity the log severity.
+ * @param nargs the number of variable parameters passed to the format string.
+ * @param log_ptr pointer (address) of the log string.
  * @param format a format string, as described in print.h. This must be a string
  * literal.
  * @param ... format parameters matching the format string.
  */
-void base_log_internal_dv(log_severity_t severity, const char *format, ...) {
-  // Do nothing, for now.
-}
\ No newline at end of file
+void base_log_internal_dv(uint32_t nargs, uintptr_t log_ptr, const char *format, ...) {
+  va_list args;
+  va_start(args, format);
+  mmio_region_write32(kSwLogAddrDv, 0x0, log_ptr);
+  for (uint32_t i = 0; i < nargs; i++) {
```

Comment:
On RISC-V, a va_list is just a void* to the stack. Why not just assume you always have smallnum (16, 32) args? Everything after the last "real" argument will probably be garbage (since you are reading all arguments as uint32_t, you're already committing UB).

Also, prefer `int` and `++i` for loops.